src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -145,19 +142,16 @@ applyCmd
       -- ^ command with payload to execute
     -> ModuleCache
       -- ^ cached module state
-    -> Bool
-      -- ^ execution config for module install
     -> IO (T2 (CommandResult [TxLog Value]) ModuleCache)
-applyCmd v logger pdbenv miner gasModel pd spv cmdIn mcache0 ecMod =
+applyCmd v logger pdbenv miner gasModel pd spv cmdIn mcache0 =
     second _txCache <$!>
       runTransactionM cenv txst applyBuyGas
   where
     txst = TransactionState mcache0 mempty 0 Nothing (_geGasModel freeGasEnv)
 
-    executionConfigNoHistory = mkExecutionConfig $
-      [ FlagDisableHistoryInTransactionalMode ] ++
-      [ FlagDisableModuleInstall | not ecMod ] ++
-      [ FlagOldReadOnlyBehavior | isPactBackCompatV16 ]
+    executionConfigNoHistory = mkExecutionConfig
+      $ FlagDisableHistoryInTransactionalMode
+      : [ FlagOldReadOnlyBehavior | isPactBackCompatV16 ]
```

Comment:
is this... a list comprehension? I've never seen that style before!

Reply:
yes, list comprehension are desugared to `MonadPlus` computations. A boolean expression `exp` becomes `guard exp`. It's quite useful to filter the results of the comprehension.

Reply:
I switched to comprehensions in a recent PR, but the credit goes entirely to Dr Kuhtz (used in block header validation)